File: src/main/java/com/youtube/tracker/models/ResponseWrapper.java
package com.youtube.tracker.models;

import com.youtube.tracker.constants.ResultInfoConstants;
import lombok.Value;

@Value
public class ResponseWrapper<T> {
    ResultInfo resultInfo;
    T data;

    public static <T> ResponseWrapper<T> success(T data) {
        return new ResponseWrapper(ResultInfoConstants.SUCCESS, data);
    }

    public static <T> ResponseWrapper<T> success() {
        return new ResponseWrapper(ResultInfoConstants.SUCCESS, null);
    }
}

File: src/main/kotlin/com/youtube/tracker/models/ResponseWrapper.kt
package com.youtube.tracker.models

import com.youtube.tracker.constants.ResultInfoConstants

data class ResponseWrapper<T>(
    val resultInfo: ResultInfo,
    val data: T?
) {
    companion object {
        fun <T> success(data: T? = null) = ResponseWrapper(ResultInfoConstants.SUCCESS, data)
    }
}